test/encryption/encryption.test.js

```diff
@@ -852,6 +852,55 @@ describe('encryption integration tests', () => {
         });
       });
 
+      describe('cloned parent schema before declaring discriminator', function() {
+        beforeEach(async function() {
+          connection = createConnection();
+        });
+        describe('csfle', function() {
+          it('throws on duplicate keys declared on different discriminators', async function() {
+            const schema = new Schema({
+              name: {
+                type: String, encrypt: { keyId: [keyId], algorithm }
+              }
+            }, {
+              encryptionType: 'csfle'
+            });
+            model = connection.model('Schema', schema);
+
+            assert.throws(() => {
+              const clonedSchema = schema.clone().add({
```

Comment:
In an ideal world, we would support this behavior because there are a number of users that rely on this pattern. However, we can consider throwing an error when using cloned schemas for `disctiminator()` a simplifying assumption, and add support for it later if there is demand.